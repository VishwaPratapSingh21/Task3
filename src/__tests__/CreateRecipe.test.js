import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CreateRecipe from '../components/recipes/CreateRecipes';

describe('CreateRecipe component', () => {
  const wrapper = shallow(<CreateRecipe match={{ params: { id: 1 } }} />);
  const preventDefault = jest.fn();

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('has initial state', () => {
    expect(wrapper.state().title).toEqual('');
    expect(wrapper.state().description).toEqual('');
  });

  it('updates state', () => {
    wrapper.setState({ title: 'lunch', description: 'cook' });
    expect(wrapper.find('[name="title"]').props().value).toEqual('lunch');
    expect(wrapper.find('[name="description"]').props().value).toEqual('cook');
  });

  it('has the correct form fields', () => {
    expect(wrapper.find('[name="title"]')).toHaveLength(1);
    expect(wrapper.find('[name="description"]')).toHaveLength(1);
  });

  it('has component in divs', () => {
    expect(wrapper.find('div')).toHaveLength(10);
  });

  it('renders component in spans', () => {
    expect(wrapper.find('span')).toHaveLength(3);
  });

  it('has input field', () => {
    expect(wrapper.find('input')).toHaveLength(2);
  });

  it('has breaks br field', () => {
    expect(wrapper.find('br')).toHaveLength(1);
  });

  it('renders a form', () => {
    expect(wrapper.find('form')).toHaveLength(1);
    expect(wrapper.find('form').simulate('submit', { preventDefault }));
    expect(preventDefault).toBeCalled();
    expect(wrapper.find('[name="title"]').simulate('change', { target: { name: 'title', value: 'hi' } }));
  });
});
