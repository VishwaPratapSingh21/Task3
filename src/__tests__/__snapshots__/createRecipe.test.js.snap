// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateRecipe component renders properly 1`] = `
<div>
  <button
    className="btn btn-primary"
    data-target="#addRecipeModal"
    data-toggle="modal"
    type="button"
  >
    <i
      className="fa fa-plus"
    />
     
    Add Recipe
  </button>
  <div
    id="recipesbar"
  >
    <div
      aria-hidden="true"
      aria-labelledby="exampleModalLongTitle"
      className="modal fade"
      id="addRecipeModal"
      role="dialog"
      tabIndex="-1"
    >
      <div
        className="modal-dialog"
        role="document"
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header"
          >
            <h5
              className="modal-title"
              id="exampleModalLongTitle"
            >
              Add recipes to 
              <strong />
            </h5>
            <button
              aria-label="Close"
              className="close"
              data-dismiss="modal"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
            </button>
          </div>
          <form
            onSubmit={[Function]}
          >
            <div
              className="modal-body"
            >
              <div
                className="input-group"
              >
                <span
                  className="input-group-addon pr-4"
                >
                  <i
                    className="fa fa-pencil"
                  />
                </span>
                <input
                  className="form-control"
                  name="title"
                  onChange={[Function]}
                  placeholder="recipe title"
                  type="text"
                  value=""
                />
              </div>
              <br />
              <div
                className="input-group"
              >
                <span
                  className="input-group-addon pr-4"
                >
                  <i
                    className="fa fa-list-alt"
                  />
                </span>
                <textarea
                  className="form-control description"
                  cols="50"
                  name="description"
                  onChange={[Function]}
                  placeholder="description or instructions"
                  rows="4"
                  value=""
                />
              </div>
            </div>
            <div
              className="modal-footer"
            >
              <button
                className="btn btn-secondary"
                data-dismiss="modal"
                id="closeRecipeModal"
                type="button"
              >
                Close
              </button>
              <input
                className="btn btn-primary"
                type="submit"
                value="Add Recipe"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
